oak_functions/loader/src/server.rs

```diff
@@ -116,14 +116,11 @@ impl FixedSizeBodyPadder for Response {
 
 /// Trait for implementing extensions, to implement new native functionality.
 pub trait OakApiNativeExtension {
-    /// Similar to `invoke_index` in [`wasmi::Externals`], but may return a result to be
-    /// written into the memory of the `WasmState`.
-    fn invoke(
-        &mut self,
-        wasm_state: &mut WasmState,
-        args: wasmi::RuntimeArgs,
-        request: Vec<u8>,
-    ) -> Result<Result<(), OakStatus>, wasmi::Trap>;
+    /// Invokes the extension with the given request and optionally returns a result. If no result
+    /// is expected, it returns None.  An error within the extension is reflected in the
+    /// `OakStatus`.
+    /// TODO(#2701): Stop returning OakStatus for errors internal to extensions.
+    fn invoke(&mut self, request: Vec<u8>) -> Result<Option<Vec<u8>>, OakStatus>;
```

Comment:
I'm not sure I understand the motivation of the `Result<Option<...>>` nesting. e.g. here it seems it should just return an empty vector, why add another layer of optionality?
I mean, by symmetry then you should make the request also an Option in case there is no request to be passed in, but that's already covered by passing an empty vec as input.